### ISSUE TITLE
bounce.py should conform to docstring standards

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6652]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)| @staceysern reported|
|-|-|
|Trac ID|trac#6652|
|Type|enhancement|
|Created|2013-07-28 19:36:28Z|
|Branch|https://github.com/twisted/twisted/tree/bounce-doc-6652|

bounce.py needs module and function docstrings.

<details><summary>Searchable metadata</summary>

```
trac-id__6652 6652
type__enhancement enhancement
reporter__shira shira
priority__normal normal
milestone__ 
branch__branches_bounce_doc_6652 branches/bounce-doc-6652
branch_author__shira shira
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__documentation documentation
time__1375040188000000 1375040188000000
changetime__1379887467000000 1379887467000000
version__None None
owner__shira shira

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_1"></a>|@staceysern commented|
|-|-|

(In [39369]) Branching to 'bounce-doc-6652'

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_2"></a>|@staceysern commented|
|-|-|

(In [39370]) Fixed docstrings.

refs [#6652](https://github.com/twisted/twisted/issues/6652)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)|@staceysern removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)|@staceysern set owner to @staceysern|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_5"></a>|@staceysern commented|
|-|-|

(In [39779]) Revised API documentation

refs [#6652](https://github.com/twisted/twisted/issues/6652)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_6"></a>|@staceysern removed owner|
|-|-|

Revised to incorporate comments from other mail documentation code reviews and for general improvement in readability. Revisions include:

 * Changing type references from str to bytes
 * Removing (optional) in param descriptions
 * Changing the format of type descriptions of tuples

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl set owner to @staceysern|
|-|-|

Hi shira, thanks for working on this.

This one looks quite open and shut to me - the docstrings make sense, they do as they say, and my twistedchecker comes up green. twisted/topfiles/6652.doc does have API in lowercase, which I think should probably be changed to match twisted/mail/topfiles/6652.doc, though.

Once that's fixed, +1 for merging.

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_9"></a>|@staceysern commented|
|-|-|

(In [40052]) Remove extraneous topfile

refs [#6652](https://github.com/twisted/twisted/issues/6652)

**Comment by twisted-trac** [comment]
|[<img alt="staceysern's avatar" src="https://avatars.githubusercontent.com/u/3631162?s=50" width="50" height="50">](https://github.com/staceysern)<a name="note_10"></a>|@staceysern set status to `closed`|
|-|-|

(In [40053]) Merge bounce-doc-6652: Fix docstrings for bounce.py

Author: shira
Reviewer: hawkowl
Fixes: [#6652](https://github.com/twisted/twisted/issues/6652)
